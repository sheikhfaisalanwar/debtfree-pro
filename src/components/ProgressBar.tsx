import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

interface ProgressBarProps {
  progress: number; // 0-100
  label?: string;
  showPercentage?: boolean;
}

export const ProgressBar: React.FC<ProgressBarProps> = ({
  progress,
  label,
  showPercentage = true,
}) => {
  const clampedProgress = Math.max(0, Math.min(100, progress));
  
  return (
    <View style={styles.container}>
      {label && (
        <View style={styles.labelRow}>
          <Text style={styles.label}>{label}</Text>
          {showPercentage && (
            <Text style={styles.percentage}>{clampedProgress.toFixed(0)}% Complete</Text>
          )}
        </View>
      )}
      <View style={styles.track}>
        <View style={[styles.fill, { width: `${clampedProgress}%` }]} />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginBottom: 16,
  },
  labelRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 8,
  },
  label: {
    fontSize: 16,
    fontWeight: '600',
    color: '#1f2937',
  },
  percentage: {
    fontSize: 14,
    color: '#6b7280',
  },
  track: {
    height: 8,
    backgroundColor: '#e5e7eb',
    borderRadius: 4,
    overflow: 'hidden',
  },
  fill: {
    height: '100%',
    backgroundColor: '#10b981',
    borderRadius: 4,
  },
});