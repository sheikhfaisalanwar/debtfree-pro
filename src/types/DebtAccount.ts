export interface DebtAccount {
  id: string;
  name: string;
  type: DebtAccountType;
  institution: string;
  accountNumber?: string;
  dueDate?: number; // Day of month
  createdDate: Date;
  lastUpdated: Date;
}

export enum DebtAccountType {
  CREDIT_CARD = 'credit_card',
  AUTO_LOAN = 'auto_loan',
  PERSONAL_LOAN = 'personal_loan',
  LINE_OF_CREDIT = 'line_of_credit',
  STUDENT_LOAN = 'student_loan',
  MORTGAGE = 'mortgage',
  OTHER = 'other'
}

export interface CreateDebtAccountParams {
  name: string;
  type: DebtAccountType;
  institution: string;
  accountNumber?: string;
  dueDate?: number;
}

export interface UpdateDebtAccountParams {
  id: string;
  name?: string;
  type?: DebtAccountType;
  institution?: string;
  accountNumber?: string;
  dueDate?: number;
}