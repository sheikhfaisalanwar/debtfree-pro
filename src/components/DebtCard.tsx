import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import { DebtAccount, DebtAccountType } from '../types/DebtAccount';
import { DebtBalance } from '../types/DebtBalance';

interface DebtCardProps {
  account: DebtAccount;
  balance: DebtBalance;
  isPriority?: boolean;
  onPress?: () => void;
}

export const DebtCard: React.FC<DebtCardProps> = ({ account, balance, isPriority, onPress }) => {
  const getDebtTypeIcon = (type: DebtAccountType): string => {
    switch (type) {
      case DebtAccountType.CREDIT_CARD:
        return '💳';
      case DebtAccountType.AUTO_LOAN:
        return '🚗';
      case DebtAccountType.PERSONAL_LOAN:
        return '💰';
      case DebtAccountType.LINE_OF_CREDIT:
        return '📈';
      case DebtAccountType.STUDENT_LOAN:
        return '🎓';
      case DebtAccountType.MORTGAGE:
        return '🏠';
      default: 
        return '📄';
    }
  };

  const formatCurrency = (amount: number): string => {
    return new Intl.NumberFormat('en-CA', {
      style: 'currency',
      currency: 'CAD',
    }).format(amount);
  };

  return (
    <TouchableOpacity
      style={[styles.container, isPriority && styles.priorityContainer]}
      onPress={onPress}
    >
      <View style={styles.header}>
        <View style={styles.titleRow}>
          <Text style={styles.icon}>{getDebtTypeIcon(account.type)}</Text>
          <Text style={styles.name}>{account.name}</Text>
          {isPriority && <View style={styles.priorityBadge}>
            <Text style={styles.priorityText}>FOCUS</Text>
          </View>}
        </View>
        <Text style={styles.balance}>{formatCurrency(balance.balance)}</Text>
      </View>
      
      <View style={styles.details}>
        <Text style={styles.detailText}>
          {balance.interestRate.toFixed(2)}% APR • Min: {formatCurrency(balance.minimumPayment)}/month
        </Text>
        {account.institution && (
          <Text style={styles.institution}>{account.institution}</Text>
        )}
        {balance.creditLimit && (
          <Text style={styles.creditInfo}>
            Credit Limit: {formatCurrency(balance.creditLimit)}
            {balance.availableCredit && ` • Available: ${formatCurrency(balance.availableCredit)}`}
          </Text>
        )}
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    borderRadius: 12,
    padding: 16,
    marginBottom: 12,
    borderWidth: 1,
    borderColor: '#e5e7eb',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
  },
  priorityContainer: {
    borderColor: '#ef4444',
    backgroundColor: 'rgba(239, 68, 68, 0.05)',
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    marginBottom: 8,
  },
  titleRow: {
    flexDirection: 'row',
    alignItems: 'center',
    flex: 1,
  },
  icon: {
    fontSize: 20,
    marginRight: 8,
  },
  name: {
    fontSize: 16,
    fontWeight: '600',
    color: '#1f2937',
    flex: 1,
  },
  balance: {
    fontSize: 18,
    fontWeight: '700',
    color: '#1f2937',
  },
  priorityBadge: {
    backgroundColor: '#ef4444',
    paddingHorizontal: 8,
    paddingVertical: 4,
    borderRadius: 12,
    marginLeft: 8,
  },
  priorityText: {
    color: 'white',
    fontSize: 10,
    fontWeight: '600',
  },
  details: {
    marginTop: 4,
  },
  detailText: {
    fontSize: 14,
    color: '#6b7280',
  },
  institution: {
    fontSize: 12,
    color: '#9ca3af',
    marginTop: 2,
  },
  creditInfo: {
    fontSize: 12,
    color: '#9ca3af',
    marginTop: 2,
  },
});