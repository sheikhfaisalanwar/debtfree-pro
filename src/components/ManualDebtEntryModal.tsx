import React, { useState } from 'react';
import {
  View,
  Text,
  Modal,
  TextInput,
  TouchableOpacity,
  StyleSheet,
  Alert,
  ScrollView,
  KeyboardAvoidingView,
  Platform,
} from 'react-native';
import { Debt, DebtType } from '../types/Debt';
import { DataStoreService } from '../services/DataStoreService';

interface ManualDebtEntryModalProps {
  visible: boolean;
  onClose: () => void;
  onDebtCreated: (debt: Debt) => void;
  fileName?: string;
}

export const ManualDebtEntryModal: React.FC<ManualDebtEntryModalProps> = ({
  visible,
  onClose,
  onDebtCreated,
  fileName,
}) => {
  const [name, setName] = useState('');
  const [type, setType] = useState<DebtType>(DebtType.CREDIT_CARD);
  const [balance, setBalance] = useState('');
  const [minimumPayment, setMinimumPayment] = useState('');
  const [interestRate, setInterestRate] = useState('');
  const [institution, setInstitution] = useState('');
  const [accountNumber, setAccountNumber] = useState('');
  const [dueDate, setDueDate] = useState('');
  const [loading, setLoading] = useState(false);

  const resetForm = () => {
    setName('');
    setType(DebtType.CREDIT_CARD);
    setBalance('');
    setMinimumPayment('');
    setInterestRate('');
    setInstitution('');
    setAccountNumber('');
    setDueDate('');
  };

  const handleClose = () => {
    resetForm();
    onClose();
  };

  const validateForm = (): string | null => {
    if (!name.trim()) return 'Debt name is required';
    if (!balance.trim()) return 'Balance is required';
    if (!minimumPayment.trim()) return 'Minimum payment is required';
    if (!interestRate.trim()) return 'Interest rate is required';

    const balanceNum = parseFloat(balance);
    const minPaymentNum = parseFloat(minimumPayment);
    const interestRateNum = parseFloat(interestRate);
    const dueDateNum = dueDate ? parseInt(dueDate) : undefined;

    if (isNaN(balanceNum) || balanceNum < 0) return 'Balance must be a valid positive number';
    if (isNaN(minPaymentNum) || minPaymentNum < 0) return 'Minimum payment must be a valid positive number';
    if (isNaN(interestRateNum) || interestRateNum < 0 || interestRateNum > 100) return 'Interest rate must be between 0 and 100';
    if (dueDateNum && (dueDateNum < 1 || dueDateNum > 31)) return 'Due date must be between 1 and 31';

    return null;
  };

  const handleSave = async () => {
    const validationError = validateForm();
    if (validationError) {
      Alert.alert('Validation Error', validationError);
      return;
    }

    try {
      setLoading(true);

      const debt: Debt = {
        id: `debt_${Date.now()}_${Math.random().toString(36).substr(2, 9)}`,
        name: name.trim(),
        type,
        balance: parseFloat(balance),
        minimumPayment: parseFloat(minimumPayment),
        interestRate: parseFloat(interestRate),
        lastUpdated: new Date(),
        institution: institution.trim() || undefined,
        accountNumber: accountNumber.trim() || undefined,
        dueDate: dueDate ? parseInt(dueDate) : undefined,
      };

      await DataStoreService.addDebt(debt);
      onDebtCreated(debt);
      handleClose();

      Alert.alert(
        'Debt Added Successfully',
        `"${debt.name}" has been added to your debt portfolio. You can now upload statements to track its progress.`,
        [{ text: 'OK', style: 'default' }]
      );
    } catch (error) {
      Alert.alert(
        'Error',
        `Failed to save debt: ${error instanceof Error ? error.message : 'Unknown error'}`,
        [{ text: 'OK', style: 'default' }]
      );
    } finally {
      setLoading(false);
    }
  };

  const debtTypes = [
    { value: DebtType.CREDIT_CARD, label: 'Credit Card' },
    { value: DebtType.LINE_OF_CREDIT, label: 'Line of Credit' },
    { value: DebtType.AUTO_LOAN, label: 'Auto Loan' },
    { value: DebtType.PERSONAL_LOAN, label: 'Personal Loan' },
    { value: DebtType.STUDENT_LOAN, label: 'Student Loan' },
    { value: DebtType.MORTGAGE, label: 'Mortgage' },
    { value: DebtType.OTHER, label: 'Other' },
  ];

  return (
    <Modal
      visible={visible}
      animationType="slide"
      presentationStyle="pageSheet"
      onRequestClose={handleClose}
    >
      <KeyboardAvoidingView
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
        style={styles.container}
      >
        <View style={styles.header}>
          <TouchableOpacity onPress={handleClose} style={styles.cancelButton}>
            <Text style={styles.cancelButtonText}>Cancel</Text>
          </TouchableOpacity>
          <Text style={styles.title}>Add Debt Manually</Text>
          <TouchableOpacity
            onPress={handleSave}
            style={[styles.saveButton, loading && styles.saveButtonDisabled]}
            disabled={loading}
          >
            <Text style={styles.saveButtonText}>{loading ? 'Saving...' : 'Save'}</Text>
          </TouchableOpacity>
        </View>

        <ScrollView style={styles.content} showsVerticalScrollIndicator={false}>
          {fileName && (
            <View style={styles.infoBox}>
              <Text style={styles.infoTitle}>ðŸ“„ PDF Statement Uploaded</Text>
              <Text style={styles.infoText}>
                {fileName} was uploaded successfully, but automatic data extraction is not available.
                Please enter the debt details manually to track this account.
              </Text>
            </View>
          )}

          <View style={styles.section}>
            <Text style={styles.sectionTitle}>Basic Information</Text>
            
            <View style={styles.field}>
              <Text style={styles.label}>Debt Name *</Text>
              <TextInput
                style={styles.textInput}
                value={name}
                onChangeText={setName}
                placeholder="e.g., Chase Freedom Card, Car Loan"
                placeholderTextColor="#9ca3af"
              />
            </View>

            <View style={styles.field}>
              <Text style={styles.label}>Debt Type *</Text>
              <ScrollView horizontal showsHorizontalScrollIndicator={false} style={styles.typeContainer}>
                {debtTypes.map((debtType) => (
                  <TouchableOpacity
                    key={debtType.value}
                    style={[
                      styles.typeButton,
                      type === debtType.value && styles.typeButtonSelected,
                    ]}
                    onPress={() => setType(debtType.value)}
                  >
                    <Text
                      style={[
                        styles.typeButtonText,
                        type === debtType.value && styles.typeButtonTextSelected,
                      ]}
                    >
                      {debtType.label}
                    </Text>
                  </TouchableOpacity>
                ))}
              </ScrollView>
            </View>
          </View>

          <View style={styles.section}>
            <Text style={styles.sectionTitle}>Financial Details</Text>
            
            <View style={styles.field}>
              <Text style={styles.label}>Current Balance *</Text>
              <TextInput
                style={styles.textInput}
                value={balance}
                onChangeText={setBalance}
                placeholder="0.00"
                placeholderTextColor="#9ca3af"
                keyboardType="decimal-pad"
              />
            </View>

            <View style={styles.field}>
              <Text style={styles.label}>Minimum Payment *</Text>
              <TextInput
                style={styles.textInput}
                value={minimumPayment}
                onChangeText={setMinimumPayment}
                placeholder="0.00"
                placeholderTextColor="#9ca3af"
                keyboardType="decimal-pad"
              />
            </View>

            <View style={styles.field}>
              <Text style={styles.label}>Interest Rate (%) *</Text>
              <TextInput
                style={styles.textInput}
                value={interestRate}
                onChangeText={setInterestRate}
                placeholder="0.00"
                placeholderTextColor="#9ca3af"
                keyboardType="decimal-pad"
              />
            </View>
          </View>

          <View style={styles.section}>
            <Text style={styles.sectionTitle}>Additional Information</Text>
            
            <View style={styles.field}>
              <Text style={styles.label}>Institution</Text>
              <TextInput
                style={styles.textInput}
                value={institution}
                onChangeText={setInstitution}
                placeholder="e.g., Chase, Wells Fargo"
                placeholderTextColor="#9ca3af"
              />
            </View>

            <View style={styles.field}>
              <Text style={styles.label}>Account Number (Last 4 digits)</Text>
              <TextInput
                style={styles.textInput}
                value={accountNumber}
                onChangeText={setAccountNumber}
                placeholder="****"
                placeholderTextColor="#9ca3af"
                maxLength={4}
                keyboardType="number-pad"
              />
            </View>

            <View style={styles.field}>
              <Text style={styles.label}>Due Date (Day of Month)</Text>
              <TextInput
                style={styles.textInput}
                value={dueDate}
                onChangeText={setDueDate}
                placeholder="15"
                placeholderTextColor="#9ca3af"
                keyboardType="number-pad"
                maxLength={2}
              />
            </View>
          </View>
        </ScrollView>
      </KeyboardAvoidingView>
    </Modal>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f9fafb',
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingHorizontal: 20,
    paddingVertical: 16,
    backgroundColor: 'white',
    borderBottomWidth: 1,
    borderBottomColor: '#e5e7eb',
  },
  cancelButton: {
    paddingVertical: 8,
    paddingHorizontal: 12,
  },
  cancelButtonText: {
    fontSize: 16,
    color: '#6b7280',
  },
  title: {
    fontSize: 18,
    fontWeight: '600',
    color: '#1f2937',
  },
  saveButton: {
    backgroundColor: '#3b82f6',
    paddingVertical: 8,
    paddingHorizontal: 16,
    borderRadius: 8,
  },
  saveButtonDisabled: {
    backgroundColor: '#9ca3af',
  },
  saveButtonText: {
    fontSize: 16,
    fontWeight: '600',
    color: 'white',
  },
  content: {
    flex: 1,
    paddingHorizontal: 20,
  },
  infoBox: {
    backgroundColor: '#fef3c7',
    borderRadius: 12,
    padding: 16,
    marginVertical: 20,
    borderLeftWidth: 4,
    borderLeftColor: '#f59e0b',
  },
  infoTitle: {
    fontSize: 16,
    fontWeight: '600',
    color: '#92400e',
    marginBottom: 8,
  },
  infoText: {
    fontSize: 14,
    color: '#92400e',
    lineHeight: 20,
  },
  section: {
    backgroundColor: 'white',
    borderRadius: 12,
    padding: 20,
    marginVertical: 12,
  },
  sectionTitle: {
    fontSize: 18,
    fontWeight: '600',
    color: '#1f2937',
    marginBottom: 16,
  },
  field: {
    marginBottom: 20,
  },
  label: {
    fontSize: 14,
    fontWeight: '500',
    color: '#374151',
    marginBottom: 8,
  },
  textInput: {
    borderWidth: 1,
    borderColor: '#d1d5db',
    borderRadius: 8,
    paddingHorizontal: 12,
    paddingVertical: 12,
    fontSize: 16,
    color: '#1f2937',
    backgroundColor: 'white',
  },
  typeContainer: {
    flexDirection: 'row',
  },
  typeButton: {
    backgroundColor: '#f3f4f6',
    borderRadius: 20,
    paddingHorizontal: 16,
    paddingVertical: 8,
    marginRight: 8,
    borderWidth: 1,
    borderColor: '#e5e7eb',
  },
  typeButtonSelected: {
    backgroundColor: '#3b82f6',
    borderColor: '#3b82f6',
  },
  typeButtonText: {
    fontSize: 14,
    color: '#6b7280',
    fontWeight: '500',
  },
  typeButtonTextSelected: {
    color: 'white',
  },
});