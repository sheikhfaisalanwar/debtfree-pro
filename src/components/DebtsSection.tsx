import React from 'react';
import {
  View,
  Text,
  StyleSheet,
} from 'react-native';
import { DebtCard } from './DebtCard';
import { DebtAccountBalance } from '../types/DebtBalance';

interface DebtsSectionProps {
  accountsWithBalances: DebtAccountBalance[];
  onDebtPress: (accountBalance: DebtAccountBalance) => void;
}

export const DebtsSection: React.FC<DebtsSectionProps> = ({
  accountsWithBalances,
  onDebtPress,
}) => {
  return (
    <View style={styles.card}>
      <Text style={styles.cardTitle}>ðŸ’³ Your Debts</Text>
      {accountsWithBalances.length === 0 ? (
        <View style={styles.emptyState}>
          <Text style={styles.emptyStateIcon}>ðŸ’³</Text>
          <Text style={styles.emptyStateTitle}>No debts added yet</Text>
          <Text style={styles.emptyStateText}>
            Use the "Add Debt" button above to get started with your debt payoff journey
          </Text>
        </View>
      ) : (
        accountsWithBalances.map(({ account, balance }, index) => (
          <DebtCard
            key={account.id}
            account={account}
            balance={balance}
            isPriority={index === 0} // First debt in snowball method
            onPress={() => onDebtPress({ account, balance })}
          />
        ))
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    backgroundColor: 'white',
    margin: 16,
    marginTop: 0,
    padding: 20,
    borderRadius: 16,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 8,
    elevation: 5,
  },
  cardTitle: {
    fontSize: 20,
    fontWeight: '600',
    color: '#1f2937',
    marginBottom: 16,
  },
  emptyState: {
    alignItems: 'center',
    paddingVertical: 32,
    paddingHorizontal: 16,
  },
  emptyStateIcon: {
    fontSize: 48,
    marginBottom: 16,
  },
  emptyStateTitle: {
    fontSize: 18,
    fontWeight: '600',
    color: '#1f2937',
    marginBottom: 8,
    textAlign: 'center',
  },
  emptyStateText: {
    fontSize: 14,
    color: '#6b7280',
    textAlign: 'center',
    lineHeight: 20,
  },
});