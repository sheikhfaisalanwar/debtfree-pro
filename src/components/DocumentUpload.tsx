import React, { useState } from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  Alert,
  ActivityIndicator,
  Modal,
} from 'react-native';
import { StatementProcessingService } from '../services/StatementProcessingService';
import { DocumentManagerService } from '../services/DocumentManagerService';
import { ManualDebtEntryModal } from './ManualDebtEntryModal';

interface DocumentUploadProps {
  onSuccess?: () => void;
}

export const DocumentUpload: React.FC<DocumentUploadProps> = ({ onSuccess }) => {
  const [showActionModal, setShowActionModal] = useState(false);
  const [showManualEntryModal, setShowManualEntryModal] = useState(false);
  const [uploading, setUploading] = useState(false);
  const [lastUploadResult, setLastUploadResult] = useState<string | null>(null);
  const [pendingStatement, setPendingStatement] = useState<any>(null);

  const handleDocumentUpload = async () => {
    try {
      setUploading(true);
      setLastUploadResult(null);
      setShowActionModal(false);

      // Upload and process the document
      const uploadResult = await DocumentManagerService.uploadAndProcessDocument();
      
      if (!uploadResult.success || !uploadResult.statement) {
        throw new Error(uploadResult.error || 'Upload failed');
      }

      // Check if meaningful data was extracted from the PDF
      const hasExtractedData = (uploadResult.statement as any).hasExtractedData;
      
      if (hasExtractedData) {
        // PDF data was extracted - show success
        setLastUploadResult(`‚úÖ Document uploaded and processed successfully!`);
        
        Alert.alert(
          'Upload Successful! üéâ',
          'Statement processed and debt information updated successfully.',
          [{ text: 'OK', style: 'default' }]
        );
        
        onSuccess?.();
      } else {
        // No meaningful data extracted - show manual entry modal
        setPendingStatement(uploadResult.statement);
        setShowManualEntryModal(true);
        setLastUploadResult(`üìÑ Document uploaded. Please enter debt details manually.`);
      }
    } catch (error) {
      const errorMessage = error instanceof Error ? error.message : 'Upload failed';
      setLastUploadResult(`‚ùå ${errorMessage}`);
      
      Alert.alert(
        'Upload Failed',
        errorMessage,
        [{ text: 'OK', style: 'default' }]
      );
    } finally {
      setUploading(false);
    }
  };

  const handleDebtCreated = () => {
    setPendingStatement(null);
    setShowManualEntryModal(false);
    onSuccess?.();
  };

  const handleManualEntryModalClose = () => {
    setShowManualEntryModal(false);
    setPendingStatement(null);
  };

  const handleManualEntryOnly = () => {
    setShowActionModal(false);
    setShowManualEntryModal(true);
  };

  return (
    <>
      <TouchableOpacity
        style={styles.actionButton}
        onPress={() => setShowActionModal(true)}
        activeOpacity={0.7}
      >
        <Text style={styles.actionButtonIcon}>‚ûï</Text>
        <Text style={styles.actionButtonText}>Add Debt</Text>
      </TouchableOpacity>

      {/* Action Selection Modal */}
      <Modal
        visible={showActionModal}
        animationType="fade"
        transparent={true}
        onRequestClose={() => setShowActionModal(false)}
      >
        <View style={styles.modalOverlay}>
          <View style={styles.actionModalContainer}>
            <Text style={styles.actionModalTitle}>Add New Debt</Text>
            <Text style={styles.actionModalSubtitle}>
              Choose how you'd like to add your debt information
            </Text>

            <TouchableOpacity
              style={[styles.optionButton, uploading && styles.optionButtonDisabled]}
              onPress={handleDocumentUpload}
              disabled={uploading}
            >
              {uploading ? (
                <View style={styles.optionContent}>
                  <ActivityIndicator size="small" color="#3b82f6" />
                  <View style={styles.optionTextContainer}>
                    <Text style={styles.optionTitle}>Processing...</Text>
                    <Text style={styles.optionDescription}>Please wait</Text>
                  </View>
                </View>
              ) : (
                <View style={styles.optionContent}>
                  <Text style={styles.optionIcon}>üìÑ</Text>
                  <View style={styles.optionTextContainer}>
                    <Text style={styles.optionTitle}>Upload Statement</Text>
                    <Text style={styles.optionDescription}>
                      Upload a PDF or CSV statement to auto-extract debt info
                    </Text>
                  </View>
                </View>
              )}
            </TouchableOpacity>

            <TouchableOpacity
              style={styles.optionButton}
              onPress={handleManualEntryOnly}
            >
              <View style={styles.optionContent}>
                <Text style={styles.optionIcon}>‚úèÔ∏è</Text>
                <View style={styles.optionTextContainer}>
                  <Text style={styles.optionTitle}>Manual Entry</Text>
                  <Text style={styles.optionDescription}>
                    Enter debt information manually
                  </Text>
                </View>
              </View>
            </TouchableOpacity>

            <TouchableOpacity
              style={styles.cancelButton}
              onPress={() => setShowActionModal(false)}
            >
              <Text style={styles.cancelButtonText}>Cancel</Text>
            </TouchableOpacity>
          </View>
        </View>
      </Modal>

      {/* Manual Entry Modal */}
      <ManualDebtEntryModal
        visible={showManualEntryModal}
        onClose={handleManualEntryModalClose}
        onDebtCreated={handleDebtCreated}
        fileName={pendingStatement?.fileName}
      />

      {/* Result Display */}
      {lastUploadResult && (
        <View style={styles.resultContainer}>
          <Text style={styles.resultText}>{lastUploadResult}</Text>
        </View>
      )}
    </>
  );
};

const styles = StyleSheet.create({
  actionButton: {
    backgroundColor: '#3b82f6',
    borderRadius: 12,
    padding: 16,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    marginHorizontal: 16,
    marginVertical: 8,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
  },
  actionButtonIcon: {
    fontSize: 20,
    marginRight: 8,
  },
  actionButtonText: {
    color: '#ffffff',
    fontSize: 16,
    fontWeight: '600',
  },
  modalOverlay: {
    flex: 1,
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: 20,
  },
  actionModalContainer: {
    backgroundColor: 'white',
    borderRadius: 16,
    padding: 24,
    width: '100%',
    maxWidth: 400,
  },
  actionModalTitle: {
    fontSize: 20,
    fontWeight: '600',
    color: '#1f2937',
    textAlign: 'center',
    marginBottom: 8,
  },
  actionModalSubtitle: {
    fontSize: 14,
    color: '#6b7280',
    textAlign: 'center',
    marginBottom: 24,
    lineHeight: 20,
  },
  optionButton: {
    borderWidth: 1,
    borderColor: '#e5e7eb',
    borderRadius: 12,
    padding: 16,
    marginBottom: 12,
    backgroundColor: 'white',
  },
  optionButtonDisabled: {
    opacity: 0.6,
  },
  optionContent: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  optionIcon: {
    fontSize: 24,
    marginRight: 16,
  },
  optionTextContainer: {
    flex: 1,
  },
  optionTitle: {
    fontSize: 16,
    fontWeight: '600',
    color: '#1f2937',
    marginBottom: 4,
  },
  optionDescription: {
    fontSize: 14,
    color: '#6b7280',
    lineHeight: 18,
  },
  cancelButton: {
    marginTop: 8,
    padding: 12,
    alignItems: 'center',
  },
  cancelButtonText: {
    fontSize: 16,
    color: '#6b7280',
    fontWeight: '500',
  },
  resultContainer: {
    backgroundColor: '#f3f4f6',
    borderRadius: 8,
    padding: 12,
    marginHorizontal: 16,
    marginVertical: 8,
  },
  resultText: {
    fontSize: 14,
    color: '#374151',
    textAlign: 'center',
  },
});