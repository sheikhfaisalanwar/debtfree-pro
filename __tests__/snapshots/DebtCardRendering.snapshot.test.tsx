import React from 'react';
import renderer from 'react-test-renderer';
import { DebtCard } from '../../src/components/DebtCard';
import { DebtAccountType } from '../../src/types/DebtAccount';

describe('DebtCard Rendering Snapshot Tests', () => {
  it('should render credit card debt correctly', () => {
    const mockDebt = {
      id: '1',
      name: 'Store Credit Card',
      type: DebtAccountType.CREDIT_CARD,
      balance: 1247,
      minimumPayment: 45,
      interestRate: 24.99,
      lastUpdated: new Date('2024-01-01T00:00:00.000Z'),
      institution: 'Store Bank'
    };

    const tree = renderer.create(<DebtCard debt={mockDebt} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render priority debt with focus badge', () => {
    const mockDebt = {
      id: '2',
      name: 'RBC Credit Card',
      type: DebtAccountType.CREDIT_CARD,
      balance: 4890,
      minimumPayment: 125,
      interestRate: 19.99,
      lastUpdated: new Date('2024-01-01T00:00:00.000Z'),
      institution: 'RBC'
    };

    const tree = renderer.create(<DebtCard debt={mockDebt} isPriority={true} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render auto loan debt correctly', () => {
    const mockDebt = {
      id: '3',
      name: 'TD Auto Loan',
      type: DebtAccountType.AUTO_LOAN,
      balance: 18500,
      minimumPayment: 320,
      interestRate: 6.5,
      lastUpdated: new Date('2024-01-01T00:00:00.000Z'),
      institution: 'TD Bank'
    };

    const tree = renderer.create(<DebtCard debt={mockDebt} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should handle string type values (legacy compatibility)', () => {
    const mockDebt = {
      id: '4',
      name: 'Line of Credit',
      type: 'LINE_OF_CREDIT' as any, // Simulate string type from data
      balance: 3200,
      minimumPayment: 85,
      interestRate: 8.25,
      lastUpdated: new Date('2024-01-01T00:00:00.000Z'),
      institution: 'National Bank'
    };

    const tree = renderer.create(<DebtCard debt={mockDebt} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should handle snake_case type values (old format compatibility)', () => {
    const mockDebt = {
      id: '5',
      name: 'Student Loan',
      type: 'student_loan' as any, // Simulate old snake_case format
      balance: 15000,
      minimumPayment: 150,
      interestRate: 4.5,
      lastUpdated: new Date('2024-01-01T00:00:00.000Z'),
      institution: 'Student Aid'
    };

    const tree = renderer.create(<DebtCard debt={mockDebt} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should handle debt without institution', () => {
    const mockDebt = {
      id: '6',
      name: 'Personal Loan',
      type: DebtAccountType.PERSONAL_LOAN,
      balance: 5000,
      minimumPayment: 200,
      interestRate: 12.5,
      lastUpdated: new Date('2024-01-01T00:00:00.000Z')
      // No institution field
    };

    const tree = renderer.create(<DebtCard debt={mockDebt} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should handle unknown debt type', () => {
    const mockDebt = {
      id: '7',
      name: 'Unknown Debt',
      type: 'unknown_type' as any,
      balance: 1000,
      minimumPayment: 50,
      interestRate: 10.0,
      lastUpdated: new Date('2024-01-01T00:00:00.000Z'),
      institution: 'Unknown Bank'
    };

    const tree = renderer.create(<DebtCard debt={mockDebt} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});