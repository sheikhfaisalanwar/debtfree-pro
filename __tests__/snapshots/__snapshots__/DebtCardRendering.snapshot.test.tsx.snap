// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`DebtCard Rendering Snapshot Tests should handle debt without institution 1`] = `null`;

exports[`DebtCard Rendering Snapshot Tests should handle snake_case type values (old format compatibility) 1`] = `null`;

exports[`DebtCard Rendering Snapshot Tests should handle string type values (legacy compatibility) 1`] = `null`;

exports[`DebtCard Rendering Snapshot Tests should handle unknown debt type 1`] = `null`;

exports[`DebtCard Rendering Snapshot Tests should render auto loan debt correctly 1`] = `null`;

exports[`DebtCard Rendering Snapshot Tests should render credit card debt correctly 1`] = `null`;

exports[`DebtCard Rendering Snapshot Tests should render priority debt with focus badge 1`] = `null`;
